src/async-wrap.cc

```diff
@@ -181,6 +183,33 @@ static void Initialize(Local<Object> target,
 }
 
 
+void AsyncWrap::DestroyIdsCb(uv_idle_t* handle) {
+  Environment* env = Environment::from_destroy_ids_idle_handle(handle);
```

Comment:
There should be a HandleScope and Context::Scope here, the `env->async_hooks_destroy_function()` call below leaks a handle now.